[Feat] Wi-Fi Setup feature for remote device user

fan9704 opened the pull request:
# Wi-Fi Setup feature for portable device

## Get Started

### Add new Wi-Fi button in nav bar

<img width="1022" height="689" alt="image" src="https://github.com/user-attachments/assets/00d9e4fa-ebea-46a7-9632-0d6180d3e842" />

Here I use emoji text istead of icon image if anyone know how to replace please help me

### In Wifi Page 

#### Show your device current Wi-Fi status

<img width="882" height="1127" alt="image" src="https://github.com/user-attachments/assets/1f062edb-90ff-4cd4-8aa9-5fe9162f5a35" />

####Show your device available hot spot to connect

<img width="882" height="1125" alt="image" src="https://github.com/user-attachments/assets/4f0747cb-4af9-49f8-8411-0172c5b91b0c" />

#### Click Connect button and enter password to connect

<img width="595" height="321" alt="image" src="https://github.com/user-attachments/assets/6cdf3eb1-21f1-4733-8b8c-1ab8497e6495" />

Show connected

<img width="824" height="309" alt="image" src="https://github.com/user-attachments/assets/1bab9c22-d713-4678-b1a5-b655895f595d" />

vicwomg commented:
Thank you for your contribution. A few thoughts:

I don't think this should be in the main title bar. It is not common to change your wifi. It should just be a link under the settings page. No need for an icon.

Pywifi is not compatible with osx and it crashes launch with `NotImplementedError`. This error should be handled more gracefully. and I feel as though this is most relevant to pi users. Can you put this behind a launch flag like `--enable-wifi-configuration` which defaults to false.

fchangcelis commented:
i think this should be a part of the setting menu, because sometimes you have to change your wifi network and share it with your friends, good for access point and public wifi broadcast

vicwomg commented:
I think I’m fine with throwing a “platform not supported screen” in the
case of the error.

On second thought, No need to hide it behind a flag, but I do strongly
believe this should NOT be in the title bar. It should be in the settings
menu behind the is_admin permission. You don’t want basic users changing
the WiFi, accidentally or otherwise.

On Sun, Aug 10, 2025 at 8:53 AM FKT ***@***.***> wrote:

> *fan9704* left a comment (vicwomg/pikaraoke#526)
> <https://github.com/vicwomg/pikaraoke/pull/526#issuecomment-3172730072>
>
> i think this should be a part of the setting menu, because sometimes you
> have to change your wifi network and share it with your friends, good for
> access point and public wifi broadcast
>
> Thanks your replied @fchangcelis <https://github.com/fchangcelis> I agree
> with your point!
> But I think we deal with @vicwomg <https://github.com/vicwomg> 's mention
> at first
>
> If we find a method let OSX user can access Wi-Fi related feature I think
> we can have futher discussion.
>
> As for whether to put Wi-Fi in the title bar, I will wait and see if OSX
> support is confirmed. Otherwise, if this feature is optional, I believe
> it's easy to overlook this important and practical feature without
> carefully reviewing the project.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vicwomg/pikaraoke/pull/526#issuecomment-3172730072>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA7KXNQXJLA3VNPH4OQDXDT3M5TH5AVCNFSM6AAAAACC6ZQ2HGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTCNZSG4ZTAMBXGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

fan9704 commented:
@vicwomg I remove Wi-Fi option from nav bar and move to info.html

Before

<img width="626" height="836" alt="image" src="https://github.com/user-attachments/assets/5f5fc5f8-aa6c-472d-b1b2-4fbdf89758ca" />

After

<img width="599" height="116" alt="image" src="https://github.com/user-attachments/assets/be0a2836-f57f-4ebf-8bc9-9fe9eb89e159" />

Also support i18n

vicwomg commented:
merging into a branch to resolve the translation conflicts

fan9704 commented:
> I have moved this with the conflict resolutions to: #531
> 
> I have added some code to disable the feature for osx. However, neither Ubuntu nor windows launches the basic smoke test:
> 
> Ubuntu: `FileNotFoundError: [Errno 2] No such file or directory: '/var/run/wpa_supplicant'`
> 
> Windows: `ModuleNotFoundError: No module named 'comtypes'`
> 
> The pywifi library seems very limited in support. What did you test it on? Raspberry Pi? If so this might be a raspberry-pi-only feature. I am uncertain how useful wifi configuration from pikaraoke would be in non-pi systems since most users can just use the OS network configs to change wifi.

Test  in Windows 11

vicwomg commented:
we cannot merge this due to lack of cross platform support and outdated libraries
